### PR TITLE
Test images with and without root password

### PR DESCRIPTION
**Pull request by mnagy**
Also enhance the testing script, better cleanup.

Work is based on top of PR #9. Only last commit is original.

### CODE REVIEW

File: 5.5/mysql/bin/run-mysqld.sh [hunk 1]
```diff
@@ -1,43 +1,96 @@
...
+	[[ "${var}" =~ $mysql_password_regex ]]
+}
+
+valid_mysql_identifier "$MYSQL_USER"     || usage
+valid_mysql_password   "$MYSQL_PASSWORD" || usage
+valid_mysql_identifier "$MYSQL_DATABASE" || usage
 
-chown -R mysql:mysql /var/lib/mysql
+# Make sure env variables don't propagate to mysqld process.
+mysql_user="$MYSQL_USER" ; unset MYSQL_USER
```
Comment (mfojtik) [inline 5.5/mysql/bin/run-mysqld.sh]: are you sure the env vars don't get propagated to mysql process this way? the lowercase vars will be the same....
i thought you are going to unset them when you starting the mysql server

File: 5.5/mysql/bin/run-mysqld.sh [hunk 2]
```diff
@@ -1,43 +1,96 @@
...
+
+	echo 'Running mysql_install_db'
+	mysql_install_db --user=mysql --datadir=/var/lib/mysql
+	# TODO: Not needed with --user=mysql. However, we should
+	#       strive to make this script runnable without root privileges
+	#chown -R mysql:mysql /var/lib/mysql
+
+	# Now start mysqld and add appropriate users.
+	echo 'Starting mysqld to create users'
+	/opt/rh/mysql55/root/usr/libexec/mysqld \
```
Comment (mfojtik) [inline 5.5/mysql/bin/run-mysqld.sh]: how about adding /opt/rh/mysql55/root/usr/libexec into PATH in Dockerfile?

**Comment by mfojtik** [comment]
@mnagy good job! LGTM (see my 2 comments)

File: 5.5/mysql/bin/run-mysqld.sh (hunk 1 above)
Comment (mnagy) [inline 5.5/mysql/bin/run-mysqld.sh]: But these are not ENV vars (they are not exported). They are just ordinary bash variables, so they'll disappear on exec.
```
mnagy@ponik:~$ VAR=foo bash -c 'export' | grep VAR
declare -x VAR="foo"
mnagy@ponik:~$ VAR=foo bash -c 'VAR1=$VAR ; unset VAR ; bash -c export' | grep VAR
mnagy@ponik:~$
```

File: 5.5/mysql/bin/run-mysqld.sh (hunk 2 above)
Comment (mnagy) [inline 5.5/mysql/bin/run-mysqld.sh]: Just because of one or two execs? No.. It's libexec, it shouldn't be in PATH, IMO.

**Comment by mnagy** [comment]
Squashed as per @mfojtik's request.

**Comment by mfojtik** [comment]
LGTM

**Pull request merged**